src/Microsoft.AspNet.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -438,27 +428,60 @@ private void RenderUnboundAttribute(string attributeName, Chunk attributeValueCh
             }
             else
             {
-                _writer
-                    .WriteStartInstanceMethodInvocation(
-                        ExecutionContextVariableName,
-                        _tagHelperContext.ExecutionContextAddHtmlAttributeMethodName)
-                    .WriteStringLiteral(attributeName)
-                    .WriteParameterSeparator()
-                    .WriteStartMethodInvocation(_tagHelperContext.MarkAsHtmlEncodedMethodName);
+                string textValue = null;
+                var isPlainTextValue = TryGetPlainTextValue(attributeValueChunk, out textValue);
 
-                // If it's a plain text value then we need to surround the value with quotes.
                 if (isPlainTextValue)
                 {
-                    _writer.WriteStringLiteral(textValue);
+                    // If it's a plain text value then we need to surround the value with quotes.
+                    _writer
+                        .WriteStartInstanceMethodInvocation(
+                            ExecutionContextVariableName,
+                            _tagHelperContext.ExecutionContextAddHtmlAttributeMethodName)
+                        .WriteStringLiteral(attributeName)
+                        .WriteParameterSeparator()
+                        .WriteStartMethodInvocation(_tagHelperContext.MarkAsHtmlEncodedMethodName)
+                        .WriteStringLiteral(textValue)
+                        .WriteEndMethodInvocation(endLine: false)
+                        .WriteEndMethodInvocation();
+                }
+                else if (IsDynamicAttributeValue(attributeValueChunk))
+                {
+                    // Dynamic attribute value should be run through the conditional attribute removal system. It's
+                    // unbound and contains C#.
+
```

Comment:
Intentional. The comment is for the block, don't want it to be directly associated with the code below it :smile: